blocks/api/serializer.js

```diff
@@ -111,6 +111,11 @@ export function serializeAttributes( attrs ) {
 
 export function serializeBlock( block ) {
 	const blockName = block.name;
+
+	if ( 'core/nextpage' === blockName ) {
+		return '<!--nextpage-->';
+	}
+
```

Comment:
would probably be nice to keep this next to the `more` tag. if it's up here to save on computation I would suggest that there's not much to save on this one special tag over the cost of serializing the document.